draft-ietf-quic-http.md

```diff
@@ -1093,10 +1094,15 @@ error of type HTTP_FRAME_UNEXPECTED.
 
 The CANCEL_PUSH frame carries a Push ID encoded as a variable-length integer.
 The Push ID identifies the server push that is being cancelled (see
-{{frame-push-promise}}).
+{{frame-push-promise}}).  If a CANCEL_PUSH frame is received which references a
+Push ID greater than currently allowed on the connection, this MUST be treated
```

Comment:
Should this only be enforced by the client, given it's the one limiting the number of pushes?  The next paragraph discusses the server closing the connection when the client sends a CANCEL_PUSH mentioning a push ID that's never been used by the server.

Reply:
I think there are two effective limits - the maximum allowed push ID and the maximum used push ID. So if I understand Ian's point, there is not a situation where a client can issue a CANCEL_PUSH that violates the maximum allowed push ID without first triggering the violation of the maximum used push ID.

Reply:
The server can issue a CANCEL_PUSH for a push it wasn't allowed to send, which the client should treat as an error.  And since CANCEL_PUSH and MAX_PUSH_ID are on the control stream, if the client uses a Push ID greater than it has permitted, that also seems invalid.

However, in the latter case, I can believe there are implementations where that ordering isn't maintained within the client, so perhaps we should relax that one.

Reply:
SG, I think I'm ok with this then.  I still find this section a bit of a hard read, but I don't have any specific suggestions.